spec/rubocop/cop/style/escaped_quotes_spec.rb

```diff
@@ -0,0 +1,316 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/RedundantStringEscape
+RSpec.describe RuboCop::Cop::Style::EscapedQuotes, :config do
+  let(:cop_config) { { 'EnforcedStyle' => style } }
+  let(:other_cops) do
+    {
+      'Style/PercentLiteralDelimiters' => {
+        'Enabled' => true,
+        'PreferredDelimiters' => {
+          'default' => '()'
+        }
+      }
+    }
+  end
+
+  shared_examples 'common' do
+    it 'does not register an offense for a single quoted string without quotes' do
+      expect_no_offenses(<<~RUBY)
+        'foo'
+      RUBY
+    end
+
+    it 'does not register an offense for a double quoted string without quotes' do
+      expect_no_offenses(<<~RUBY)
+        "foo"
+      RUBY
+    end
+
+    it 'does not register an offense for a single quoted string that ends with an escaped slash' do
+      expect_no_offenses(<<~RUBY)
+        'foo\\\\'
+      RUBY
+    end
+
+    it 'does not register an offense for a double quoted string that ends with an escaped slash' do
+      expect_no_offenses(<<~RUBY)
+        "foo\\\\"
+      RUBY
+    end
+
+    it 'does not register an offense for a single quoted string that contains an escape sequence' do
+      expect_no_offenses(<<~RUBY)
+        '\n'
```

Comment:
This and following don't seem sufficient.
We need to cover the case that otherwisse would be an offence. But we deliberately decide to ignore it because there's an escape sequence because the result will be different if we change the quotes. E.g. `"foo\"bar\n"`.

Reply:
I'm not following you here. The point of this test was just to ensure that other non-quote escape sequences (or in the case of this specific example, strings that look like escape sequences) do not trigger the cop. It's not deliberately ignored because of the escape sequence, it's ignored because there's no escaped quote.

There are tests for a string including both an escaped quote and another escape sequence, for example here: https://github.com/rubocop/rubocop/blob/7baa887d06671435fe4a7af31b66d6184bddaf43/spec/rubocop/cop/style/escaped_quotes_spec.rb#L280-L289